### ISSUE TITLE
win.center() will center to the wrong display when you call it right after new BrowserWindow()

### ISSUE THREAD

**Issue by jwu** [electron/electron#8114]
* Electron version: 1.14.+
* Operating system: Mac OSX

### Expected behavior

The window should center to the display it lives.

### Actual behavior

The window always center to the external display.

### How to reproduce

First you should have multi-display working environment, this is mine:

```
|-----------------------|
|                       |
|    external display   |
|                       |
|-----------------------|
    |------------|
    |   macbook  |
    |------------|
```

When I use the following code:

```js
    let primaryDisplay = electron.screen.getPrimaryDisplay();
    win1 = new BrowserWindow({
      width: 400,
      height: 300,
    });
    win1.setPosition( primaryDisplay.workArea.x + 50, primaryDisplay.workArea.y + 50 );
    win1.center();

    win1.loadURL(url.format({
      pathname: path.join(__dirname, 'index.html'),
      protocol: 'file:',
      slashes: true
    }));
```

The window didn't center to the display I want, instead it center to an external display.

I create a very simple test project to reproduce this problem: https://github.com/jwu/electron-pitfalls, just clone it and run:

```
npm start win-center
```

The code relate with this problem is in: https://github.com/jwu/electron-pitfalls/tree/master/win-center

First I think may be the `win.center()` always center to the last display. But I try to move the window in different display and **click the center button in it**, it will center to the display it currently lives in. So I think this bug only happen when you call `center()` right after you create a window.

A related issue that may be report the same problem: https://github.com/electron/electron/issues/3490

**Comment by nadav-dav** [comment]
I'll add, that when the app is in fullscreen, and it tries to center it - since it can't come out of the fullscreen area (i'm using osx, with work spaces) it just crops the app and you are left with a black screen.

**Comment by sofianguy** [comment]
Thanks for contributing!

We are no longer releasing bug fixes for versions of Electron older than 1.6x. We prefer to only keep issues that are on our immediate radar.

If you are still having the same issue in an up-to-date version, feel free to reach out and we'll reopen.
